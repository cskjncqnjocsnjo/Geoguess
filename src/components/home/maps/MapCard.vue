<template>
    <v-card width="300" rounded="lg">
        <v-img
            class="white--text align-end"
            height="140px"
            gradient="rgba(0,0,0,0), rgba(0,0,0,0.8)"
            :src="
                mapLocate.imageUrl ||
                `https://source.unsplash.com/300x140/daily?${encodeURI(
                    mapLocate.nameLocate
                )}`
            "
        >
            <v-card-title>{{ mapLocate.nameLocate }}</v-card-title>
        </v-img>
        <v-card-actions class="map-card__actions">
            <v-spacer></v-spacer>
            <MapDialog :mapLocate="mapLocate" />
        </v-card-actions>
    </v-card>
</template>

<script>
import { mapActions } from 'vuex';
import MapDialog from '@/components/home/maps/MapDialog';
export default {
    name: 'MapCard',
    components: {
        MapDialog,
    },
    props: {
        map: Object,
    },
    computed: {
        mapLocate() {
            return {
                ...this.map,
                nameLocate:
                    this.map.name[this.$i18n.locale] || this.map.name['en'],
                descriptionLocate:
                    this.map.description[this.$i18n.locale] ||
                    this.map.description['en'],
            };
        },
    },
    methods: {
        ...mapActions(['loadGeoJsonFromUrl']),
    },
};
</script>

<style lang="scss">
.map-card__actions {
    padding: 1%;
}
</style>
