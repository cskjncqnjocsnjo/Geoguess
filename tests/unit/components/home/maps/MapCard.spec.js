import MapCard from '@/components/home/maps/MapCard.vue';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import appInit from '../../../utils/appInit';

const args = appInit(createLocalVue());

describe('MapDialog.vue', () => {
    let store;

    it('open dialog hist', () => {
        const map = {
            name: {
                en: 'Biggest City',
                fr: 'Grande Villes',
            },
            description: {
                fr:
                    'Une liste de plus de 40 villes : PARIS, KOBE OSAKA, SEOUL, HOUSTON, BARCELONA, PHILADELPHIA, SANTIAGO, LAGOS, DALLAS, NEW YORK, ISTANBUL, TOKYO, FUKUOKA, LONDON, KUALA LUMPUR, LIMA, HO CHI MINH CITY, MANILA, GUADALAJARA, MADRID, NAGOYA, SINGAPORE, JOHANNESBURG, BELO HORIZONTE, TORONTO, MEXICO CITY, MIAMI, ATLANTA, RIO DE JANEIRO, BUENOS AIRES, SAO PAULO, CHICAGO,…',
                en:
                    'List of 40 biggest cities of the world: PARIS, KOBE OSAKA, SEOUL, HOUSTON, BARCELONA, PHILADELPHIA, SANTIAGO, LAGOS, DALLAS, NEW YORK, ISTANBUL, TOKYO, FUKUOKA, LONDON, KUALA LUMPUR, LIMA, HO CHI MINH CITY, MANILA, GUADALAJARA, MADRID, NAGOYA, SINGAPORE, JOHANNESBURG, BELO HORIZONTE, TORONTO, MEXICO CITY, MIAMI, ATLANTA, RIO DE JANEIRO, BUENOS AIRES, SAO PAULO, CHICAGO,…',
            },
            author: 'BilelJegham',
            imageUrl:
                'https://images.unsplash.com/photo-1477959858617-67f85cf4f1df?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=140',
            url: 'https://mapurl.geojson',
        };
        const wrapper = shallowMount(MapCard, {
            ...args,
            store,
            propsData: {
                map,
            },
        });

        const mapLocate = wrapper.vm.mapLocate;

        expect(mapLocate.descriptionLocate).toEqual(map.description.en);

        expect(mapLocate.nameLocate).toEqual(map.name.en);

        expect(wrapper).toMatchSnapshot();
    });
});
