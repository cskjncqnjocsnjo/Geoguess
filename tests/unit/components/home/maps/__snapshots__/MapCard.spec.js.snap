// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDialog.vue open dialog hist 1`] = `
<v-card-stub loaderheight="4" tag="div" width="300" rounded="lg">
  <v-img-stub height="140px" gradient="rgba(0,0,0,0), rgba(0,0,0,0.8)" options="[object Object]" position="center center" src="https://images.unsplash.com/photo-1477959858617-67f85cf4f1df?ixlib=rb-1.2.1&amp;ixid=eyJhcHBfaWQiOjEyMDd9&amp;auto=format&amp;fit=crop&amp;w=300&amp;q=140" transition="fade-transition" class="white--text align-end">
    <v-card-title-stub>Biggest City</v-card-title-stub>
  </v-img-stub>
  <v-card-actions-stub class="map-card__actions">
    <v-spacer-stub></v-spacer-stub>
    <mapdialog-stub maplocate="[object Object]"></mapdialog-stub>
  </v-card-actions-stub>
</v-card-stub>
`;
